pytest_reportportal/listener.py

```diff
@@ -53,67 +64,83 @@ def pytest_runtest_makereport(self, item):
                 loglevel='ERROR',
             )
 
+        # Defining test result
         if report.when == 'setup':
             self.result = None
             self.issue = {}
-            if report.failed:
-                # This happens for example when a fixture fails to run
-                # causing the test to error
+
+        if report.failed:
+            if (self.result in (None, 'PASSED', 'SKIPPED')):
                 self.result = 'FAILED'
-                self._add_issue_info(item, report)
-            elif report.skipped:
-                # This happens when a testcase is marked "skip".  It will
-                # show in reportportal as not requiring investigation.
+        elif report.skipped:
+            if (self.result in (None, 'PASSED')):
                 self.result = 'SKIPPED'
-                self._add_issue_info(item, report)
+        else:
+            if (self.result is None):
+                self.result = 'PASSED'
 
-        if report.when == 'call':
-            if report.passed:
-                item_result = 'PASSED'
-            elif report.skipped:
-                item_result = 'SKIPPED'
-                self._add_issue_info(item, report)
-            else:
-                item_result = 'FAILED'
-                self._add_issue_info(item, report)
-            self.result = item_result
+        # Adding test comment and issue type
+        self._add_issue_info(item, report)
 
+    def _add_issue_id_marks(self, item):
+        """Adds marks with issue id"""
+        issue_marks = item.session.config.getini('rp_issue_marks')
+        mark_gen = pytest.mark
+        if item.session.config.getini('rp_issue_id_marks'):
+            for mark_name in issue_marks:
+                for mark in item.iter_markers(name=mark_name):
+                    if mark:
+                        issue_ids = mark.kwargs.get("issue_id", [])
+                        if not isinstance(issue_ids, list):
+                            issue_ids = [issue_ids]
+                        for issue_id in issue_ids:
+                            tag_mark = "{}:{}".format(mark.name, issue_id)
+                            try:
+                                mark_gen._markers.add(tag_mark)  # register mark in pytest,
+                            except AttributeError:               # for pytest >= 4.5.0
+                                pass
+                            item.add_marker(tag_mark)
 
     def _add_issue_info(self, item, report):
-
-        issue_type = None
-        comment = ""
+        """Adds issues description and issue_type to the test item"""
```

Comment:
This docstring does not correspond to PEP257.
D400, First line should end with a period.
D401, First line should be in imperative mood.